azure-pipelines/secure-development-tools.yml

```diff
@@ -9,3 +18,87 @@ steps:
   displayName: 'Run PoliCheck'
   inputs:
     targetType: F
+
+ ### Analyze managed and unmanaged binaries (exe, dll) for security vulnerabilities
+- task: BinSkim@3
+  displayName: Run BinSkim
+  inputs:
+    InputType: 'Basic'
+    Function: 'analyze'
+    AnalyzeTarget: '$(Build.SourcesDirectory)/bin/StreamJsonRpc/$(BuildConfiguration)/netstandard2.0/StreamJsonRpc.dll;$(Build.SourcesDirectory)/bin/StreamJsonRpc/$(BuildConfiguration)/netcoreapp2.1/StreamJsonRpc.dll'
+
+ ### Copy existing Roslyn Analyzer logs to be analyzed
+- task: RoslynAnalyzers@3
+  displayName: Run Roslyn Analyzers.
+  inputs:
+    copyLogsOnly: 'true'
+  continueOnError: True
+
+ ### Copy files for APIScan
+- task: CopyFiles@2
+  displayName: 'Copy Files for APIScan'
+  inputs:
+    SourceFolder: '$(Build.SourcesDirectory)/bin/StreamJsonRpc/$(BuildConfiguration)'
+    Contents: |
+      netcoreapp2.1\StreamJsonRpc.*
+    TargetFolder: $(Agent.TempDirectory)\APIScanFiles\netcoreapp2.1
+  condition: ne(variables['DisableAPIScan'], 'true')
+
+- task: CopyFiles@2
+  displayName: 'Copy Files for APIScan'
+  inputs:
+    SourceFolder: '$(Build.SourcesDirectory)/bin/StreamJsonRpc/$(BuildConfiguration)'
+    Contents: |
+      netstandard2.0\StreamJsonRpc.*
+    TargetFolder: $(Agent.TempDirectory)\APIScanFiles\netstandard2.0
+  condition: ne(variables['DisableAPIScan'], 'true')
+
+ ### Run APIScan
+- task: APIScan@1
+  displayName: Run APIScan
+  inputs:
+    softwareFolder: $(Agent.TempDirectory)\APIScanFiles
+    softwareName: 'Microsoft.VisualStudio.LanguageServer'
+    softwareVersionNum: '$(Build.BuildId)'
+    isLargeApp: false
+    toolVersion: LatestPreRelease
+  continueOnError: true
+  condition: ne(variables['DisableAPIScan'], 'true')
+
+ ### Create formatted HTML code analysis log file.
+- task: securedevelopmentteam.vss-secure-development-tools.build-task-report.SdtReport@1
+  displayName: 'Create Security Analysis Report'
+  inputs:
+    BinSkim: true
+    BinSkimBreakOn: WarningAbove
+    CredScan: true
+    PoliCheck: true
+    PoliCheckBreakOn: Severity4Above
+  continueOnError: true
+
+ ### Publish Code Analysis results to build Artifacts.
+- task: securedevelopmentteam.vss-secure-development-tools.build-task-publishsecurityanalysislogs.PublishSecurityAnalysisLogs@2
+  displayName: 'Publish Code Analysis Logs'
+
+ ### Fails the build if tools generate errors
+- task: PostAnalysis@1
+  inputs:
+    AllTools: false
+    APIScan: true
+    BinSkim: true
+    BinSkimBreakOn: 'WarningAbove'
+    CodesignValidation: true
+    CodesignValidationBreakOn: 'WarningAbove'
+    CredScan: true
+    FortifySCA: false
+    FxCop: false
+    ModernCop: false
+    PoliCheck: true
+    PoliCheckBreakOn: 'Severity4Above'
+    RoslynAnalyzers: true
+    RoslynAnalyzersBreakOn: 'WarningAbove'
+    SDLNativeRules: false
+    Semmle: false
+    TSLint: false
+    ToolLogsNotFoundAction: 'Standard'
+  condition: eq(variables['BreakOnError'], 'true')
```

Comment:
This looks like the default will be that failures will not break the pipeline. Can we instead make sure we pass cleanly, and then shut the door to silent regressions by ensuring any block the pipeline immediately?

Reply:
I meant to configure BreakOnError = true as the default configuration of the pipeline and leave setting it to false as an option, but I can remove this.